src/viam/sdk/resource/resource_manager.hpp

```diff
@@ -27,6 +28,15 @@ class ResourceManager {
     /// @throws `std::runtime_error` if the desired resource does not exist.
     std::shared_ptr<Resource> resource(const std::string& name);
 
+    /// @brief Returns a resource after dynamically downcasting to `T`.
+    /// @param name of the desired resource.
+    /// @throws `std::runtime_error` if the desired resource does not exist.
+    template <typename T>
+    std::shared_ptr<T> resource(const std::string& name) {
```

Comment:
Nice; I wonder if there could be any interaction between this and the templated [`resource_by_name`](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/robot/client.hpp#L91) method on the robot client.

Reply:
Not sure quite what you mean by "interaction", can you explain a little further what you have in mind?

Reply:
I _think_ `resource_by_name` could just call this method; no need to do it here (if it's possible), just wanted to note that I've seen a similar method lying around.